src/arrayMethodJoin.js

```diff
@@ -4,9 +4,81 @@
  * Implement method join
  */
 function applyCustomJoin() {
+  let result = '';
+
   [].__proto__.join2 = function(separator) {
```

Comment:
```suggestion
  [].__proto__.join2 = function(separator = ',') {
```
just add default separator when it is not passed